utils/mermaid.js

```diff
@@ -0,0 +1,192 @@
+	let start = new Boolean(false);
+	for (let i = 0; i < callFlow.length; i++) {
+		if (start == false && Number(startId[i]) == id) {
+			start = true;
+		}
+		if (start == true) {
+			if (Number(startId[i]) == id || startDepth[i] > depth) {
+				if (open)
+					callFlow[i] = COMMENT + callFlow[i];
+				else
+					callFlow[i] = callFlow[i].replace(COMMENT, "");
+			}
+			else {
+				break;
+			}
+		}
+	}
+	if (open)
+		openNode.pop(id);
+	else
+		openNode.push(id);
+
+	renderGraph("".concat(GRAPH_TITLE, callFlow.join("\n"), "\n", showHideDesc));
+};
+
+var selectSubGraph = function(nodeId) {
+	const nodeIdResult = nodeId.match(FUNC_ID_PATTERN);
+	const id = nodeIdResult.groups.node_id;
+
+	for (let i = 0; i < callFlow.length; i++) {
+		if (Number(startId[i]) == id) {
+			drawBaseGraph(id);
+			break;
+		}
+	}
+};
+
+var initializeBaseGraph = function() {
+	startId.length = 0;
+	startDepth.length = 0;
+	startName.length = 0;
+	openNode.length = 0;
+	callFlow.length = 0;
+	showHideDesc = "";
+	baseGraph = "";
+};
+
+var drawDepthGraph = function(depth) {
+	mermaid.initialize(config);
+	showHideDesc = "";
+	openNode.length = 0;
+	const nodeList = new Array();
+	let id = new String();
+	for (let i = 0; i < callFlow.length; i++) {
+		callFlow[i] = callFlow[i].replace(COMMENT, "");
+		if (Number(startDepth[i]) >= depth)
+			callFlow[i] = (COMMENT + callFlow[i]);
+		if (Number(startDepth[i]) == depth) {
+			id = startDepth[i] + "_" + startId[i];
+			if (nodeList.includes(id) == false) {
+				nodeList.push(id);
+				showHideDesc = showHideDesc.concat(
+					"  style " + id +
+					" fill:#f9f,stroke:#333,stroke-width:px;\n");
+				showHideDesc =
+					showHideDesc.concat("  click " + id + " showHideNode;\n");
+			}
+		}
+	}
+	renderGraph("".concat(GRAPH_TITLE, callFlow.join("\n"), "\n", showHideDesc));
+};
+
+var drawSelectGraph = function() {
+	let selectionDesc = new String();
+	let nodeList = new Array();
+	if (callFlow.length == 0)
+		drawBaseGraph(0);
+	for (let i = 1; i < callFlow.length; i++) {
+		let id = startDepth[i] + "_" + startId[i];
+		if (nodeList.includes(id) == false) {
+			nodeList.push(id);
+			selectionDesc = selectionDesc.concat(
+				"  style " + id + " fill:#CAB8FF,stroke:#333,stroke-width:px;\n");
+			selectionDesc =
+				selectionDesc.concat("  click " + id + " selectSubGraph;\n");
+		}
+	}
+	renderGraph(baseGraph.concat("\n", selectionDesc));
+	document.getElementById("description").innerHTML = "Click on a node to view Sub-Graph";
+};
+
+var drawBaseGraph = function(id) {
+	mermaid.initialize(config);
+	initializeBaseGraph();
+	let start = new Boolean(false);
+	maxDepth = 0;
+	var result = mermaidTxt.split(/\r?\n/);
+	for (let i = 0; i < result.length; i++) {
+		if (result[i].length == 0)
+			continue;
+		let oneCallFlow = result[i].match(
+			/\s+(?<start_depth>\d+)_(?<start_id>\d+)\[\"(?<start_name>\S+)\"\]\s+--\S+\|(?<call_num>\d+)\|\s+(?<end_depth>\d+)_(?<end_id>\d+)\[\"(?<end_name>\S+)\"];/);
```

Comment:
It's a bit hard to follow.  Could you please add a comment with examples for each case?

Reply:
I added the comment of an example.

Reply:
Thanks!